independent-projects/tools/common/src/main/java/io/quarkus/devtools/project/buildfile/BuildFile.java

```diff
@@ -0,0 +1,158 @@
+                    try {
+                        removeDependencyFromBuildFile(k);
+                        counter.increment();
+                    } catch (IOException ex) {
+                        throw new UncheckedIOException(ex);
+                    }
+                });
+        this.writeToDisk();
+        return new UninstallResult(counter.intValue());
+    }
+
+    protected abstract void addDependencyInBuildFile(AppArtifactCoords coords) throws IOException;
+
+    protected abstract void removeDependencyFromBuildFile(AppArtifactKey key) throws IOException;
+
+    protected abstract List<Dependency> getDependencies() throws IOException;
+
+    protected abstract void writeToDisk() throws IOException;
+
+    protected abstract String getProperty(String propertyName) throws IOException;
+
+    protected abstract boolean containsBOM(String groupId, String artifactId) throws IOException;
+
+    protected abstract void refreshData();
+
+    protected Path getProjectFolderPath() {
+        return projectFolderPath;
+    }
+
+    protected boolean hasProjectFile(final String fileName) throws IOException {
+        final Path filePath = projectFolderPath.resolve(fileName);
+        return Files.exists(filePath);
+    }
+
+    protected byte[] readProjectFile(final String fileName) throws IOException {
+        final Path filePath = projectFolderPath.resolve(fileName);
+        return Files.readAllBytes(filePath);
+    }
+
+    protected void writeToProjectFile(final String fileName, final byte[] content) throws IOException {
+        Files.write(projectFolderPath.resolve(fileName), content);
+    }
+
+    private boolean isQuarkusExtension(final AppArtifactKey key) {
+        // This will not always be true as the platform descriptor does not contain the list of all available extensions
+        return isDefinedInRegistry(platformDescriptor.getExtensions(), key);
+    }
+
+    private Set<AppArtifactKey> getDependenciesKeys() throws IOException {
+        return getDependencies().stream().map(Extensions::toKey).collect(Collectors.toSet());
+    }
+
+    private String extractVersion(final Dependency d) {
+        String version = d != null ? d.getVersion() : null;
+        if (version != null && version.startsWith("$")) {
+            String value = null;
+            try {
+                value = (String) this.getProperty(propertyName(version));
+                if (value != null) {
+                    return value;
+                }
+            } catch (IOException e) {
+                // ignore this error.
+            }
+        }
+        if (version != null) {
+            return version;
+        }
+        return null;
+    }
+
+    private String propertyName(final String variable) {
+        return variable.substring(2, variable.length() - 1);
+    }
+
+    public static boolean isDefinedInRegistry(List<Extension> registry, final AppArtifactKey key) {
+        return findInRegistry(registry, key).isPresent();
+    }
```

Comment:
Don't expose this method, because the Registry won't be treated as a List of Extensions in the future